upload/src/addons/SV/WarningImprovements/XF/ControllerPlugin/Warn.php

```diff
@@ -2,11 +2,38 @@
 
 namespace SV\WarningImprovements\XF\ControllerPlugin;
 
+use XF\Mvc\Entity\Entity;
+
 /**
  * Extends \XF\ControllerPlugin\Warn
  */
 class Warn extends XFCP_Warn
 {
+    public function actionWarn($contentType, Entity $content, $warnUrl, array $breadcrumbs = [])
+    {
+        $response = parent::actionWarn($contentType, $content, $warnUrl, $breadcrumbs);
+
+        if ($response instanceof \XF\Mvc\Reply\View)
+        {
+            $categoryRepo = $this->getWarningCategoryRepo();
+            $categoryTree = $categoryRepo->createCategoryTree(null, 0, true);
```

Comment:
After a review, I can't see how Warning Categories are doing filtering based off permissions.

Not implemented yet?